### DISCUSSION TITLE
[Suggestion] add extra class for all variants to avoid overriding the published components

### DISCUSSION THREAD

**Discussion by 3adeling** [livewire/flux#147] [Feature Requests]
Hey @calebporzio,

Wouldn't it be better to add a unique class for each variant in all components? This way, it would be much easier to target them through CSS, avoiding the need to modify the published components. This would also prevent conflict issues with future updates.

button/index.blade.php:

```
    ->add(match ($variant) {
        'primary' => 'primary-button',
        'filled' => 'filled-button',
        'outline' => 'outline-button',
        'danger' => 'danger-button',
        'ghost' => 'ghost-button',
        'subtle' => 'subtle-button',
    })
   ```
   
   app.css
   
   ```
   [data-flux-button].primary-button {
    @apply bg-blue-800 hover:bg-blue-900 dark:bg-white dark:hover:bg-blue-100 text-white dark:text-blue-800;
}

[data-flux-button].filled-button {
    @apply bg-blue-800/5 hover:bg-blue-800/10 dark:bg-white/10 dark:hover:bg-white/20 text-blue-800 dark:text-white;
}

[data-flux-button].outline-button {
    @apply bg-white hover:bg-blue-50 dark:bg-blue-700 dark:hover:bg-blue-600/75 text-blue-800 dark:text-white;
}
   ```
   
  Result
  
 
<img width="829" alt="image" src="https://github.com/user-attachments/assets/cda7748e-756e-4301-8ef0-640bad5c75bf">

**Comment by joshhanley** [comment]
Thanks for reporting! We have converted this issue to a feature request for now, so we can focus on fixing any bugs first.
